### PR TITLE
Fix user extension implementation

### PR DESCRIPTION
**Pull request by bngo92**
The current extension is of type Option<User>.
This is less ergonomic for handlers that already use RequireAuthorizationLayer.

### CODE REVIEW

File: axum-login/src/auth.rs [hunk 1]
```diff
@@ -128,7 +128,12 @@ where
                     auth_cx.current_user = user;
 
                     request.extensions_mut().insert(auth_cx.clone());
-                    request.extensions_mut().insert(auth_cx.current_user);
+                    request
+                        .extensions_mut()
+                        .insert(auth_cx.current_user.clone());
```
Comment (maxcountryman) [inline axum-login/src/auth.rs:133]: Is the idea in keeping this to preserve backwards compatibility?

I'm a little concerned is confusing to have `Option<User>` and `User` as extensions.

Comment (bngo92) [reply axum-login/src/auth.rs:133]: Yes, lines 131-133 are for backwards compatibility while lines 134-136 implements the extension as described in the doc examples.

I am open to dropping lines 131-133 if you're not concerned with backwards compatibility. This change should be relatively safe as the `Option<User>` extension wasn't documented.

Comment (maxcountryman) [reply axum-login/src/auth.rs:133]: I think it's safe to drop given we're pre-1.0.

Comment (bngo92) [reply axum-login/src/auth.rs]: Sounds good, I have updated the pull request.

**Comment by maxcountryman** [comment]
Thank you for putting this together!

**Pull request merged**